Framework/Kernel/src/InstrumentInfo.cpp

```diff
@@ -121,16 +121,62 @@ std::string InstrumentInfo::filePrefix(unsigned int runNumber) const {
 }
 
 /// Returns the name of the live listener
-const std::string &InstrumentInfo::liveListener() const {
-  return m_liveListener;
+std::string InstrumentInfo::liveListener(const std::string &name) const {
+  if (!hasLiveListenerInfo())
+    return "";
```

Comment:
I'm not a big fan of this kind of silent error. However, as you point out this is existing behaviour, i think it's beyond the remit of this issue to go and fix that behaviour. You could create a new Issue for this perhaps.